### DISCUSSION TITLE
Pass further data to `Content`

### DISCUSSION THREAD

**Discussion by notlee** [withastro/starlight#1980] [Feature Requests]
### What version of `starlight` are you using?

0.23.2

### What is your idea?

Hey 👋 Love Starlight, thanks for all your work on it. Would you be open to a PR which passes further Astro props through to `Content`?

[packages/starlight/index.astro#L19](https://github.com/withastro/starlight/blob/e76bf262292545625c07ff387082db5f7def173a/packages/starlight/index.astro#L19)

It would be useful for us to be able to access the current locale:

```diff
-<Page {...route}><Content frontmatter={Astro.props.entry.data} /></Page>
+<Page {...route}><Content locale={Astro.props.locale} frontmatter={Astro.props.entry.data} /></Page>
```

But perhaps passing all Astro props could be more generally helpful?

```diff
-<Page {...route}><Content frontmatter={Astro.props.entry.data} /></Page>
+<Page {...route}><Content Astro={Astro.props} frontmatter={Astro.props.entry.data} /></Page>
```

### Why is this feature necessary?

We would like to change page content conditionally based on locale within `mdx`, without duplicating content for a full page translation.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Hey @notlee! Thanks for the suggestion.

We have a bigger refactor planned to move the current route data object to be more widely available, so I think we might wait for that, which would likely unlock some of this.

For the locale, if you’re on the latest version of Starlight (v0.24), you should be able to access it via `Astro.currentLocale` in any Astro components you are using. Maybe that unlocks what you’re looking for already?

**Reply by notlee** [reply]
Thanks for the quick reply @delucis, that's super helpful 🙇

**Comment by HiDeoo** [comment]
Starting with [Starlight 0.32.0](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.32.0), [route data](https://starlight.astro.build/guides/route-data/) provides a way to read and even update all the information required to render a single page.

All the properties available on route data are listed [here](https://starlight.astro.build/reference/route-data/), e.g. `locale`, `id`, or the `entry` itself.

The [route data guide](https://starlight.astro.build/guides/route-data/) explains how to access it or how to customize it.

Considering these changes, I'm going to close this discussion.
